integration/test-integration/src/test/java/org/wso2am/micro/gw/tests/testCases/subscription/SubscriptionValidationTestCase.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2am.micro.gw.tests.testCases.subscription;
+
+import com.github.dockerjava.zerodep.shaded.org.apache.hc.core5.http.HttpStatus;
+import org.awaitility.Awaitility;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.am.integration.test.utils.bean.APILifeCycleAction;
+import org.wso2.am.integration.test.utils.bean.APIRequest;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+import org.wso2am.micro.gw.mockbackend.ResponseConstants;
+import org.wso2am.micro.gw.tests.apim.APIMLifecycleBaseTest;
+import org.wso2am.micro.gw.tests.context.MicroGWTestException;
+import org.wso2am.micro.gw.tests.util.HttpsClientRequest;
+import org.wso2am.micro.gw.tests.util.TestConstant;
+import org.wso2am.micro.gw.tests.util.Utils;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class SubscriptionValidationTestCase extends APIMLifecycleBaseTest {
+
+    private APIRequest apiRequest;
+    private String apiId;
+    private String applicationId;
+    private Map<String, String> requestHeaders;
+    private String endpointURL;
+
+    @BeforeClass(alwaysRun = true, description = "initialise the setup")
+    void setEnvironment() throws Exception {
+        super.init();
```

Comment:
Is the super.init() specific to the SubscriptionValidationTestCase, if so shall we move it to somewhere other than the APIMLifecycleBaseTest class. If it is not specific, shall we move this line to APIManagerInitializeTestCase

Reply:
This is not specific only to SubscriptionValidationTestCase. All the apim related test cases need to call the super.init() function to initialize the context required for that particular test case. This call needs to be done in each such test cases as some test cases might need to initialize the context for a particular user. For that purpose, they can use `void init(TestUserMode userMode)`.

Also, this method cannot be moved to APIManagerInitializeTestCase since there is no way we can maintain the initialized variables throughout the test suite.

Reply:
If there can be some test cases might need to initialize the context for a particular user then it shouldn't be common to the suit right?

Reply:
The reason to ask this is because both the subscription related test case and the initialization test case extends the same APIMLifecycleBaseTest class. Was thinking whether the two stages can be separated into two classes because they'd anyway be two different instances of the same super class.

Reply:
> If there can be some test cases might need to initialize the context for a particular user then it shouldn't be common to the suit right?

I don't think we need to categorize test suits based on the user the test case is going to use. Then, there will be unnecessary suits created just for different users

Reply:
Got it. Let's leave it as it is.